File: day1_js/day1.py
# // write a function that searches for an element in an array and returns the index
def search(arr,target):
    for i in range(len(arr)):
        if arr[i]==target:
            return i
a=search([1,2,3,4,5], 4)
print(a)
    
    
def find(arr2,tar):
    for index, value in enumerate(arr2):
        if value==tar:
            return index
    return -1
b=search([1,7,8,5,6,9,2],8)
print(b)

# // // write a function that returns the number of negative numbers in an array
def neg(arr3):
    count=0
    for i in arr3:
        if i<0:
            count=count+1
    return count
print(neg([1,-5,9,-10,5,4,-7]))

# / // write a function that returns largest number in an array
def large(arr4):
    largest=0
    for i in arr4:
        if i>=largest:
            largest=i
    return largest
print(large([1,-5,9,-10,5,4,-7]))

# palindrome 

g="mom"
upd=g.replace(" ",'').lower()
rev=upd[::-1]
if rev==upd:
    print("palindrome")
else:
    print("not palindrome")

num=122
org=num
rev=0
while num!=0:
    rev=rev*10+num%10
    num=int(num/10)

if org==rev:
    print("palindrome")
else:
    print("not palindrome")

# find the second largest in array
def seclarge(arr5):
    firstlargest=float('-inf')
    secondlargest=float('-inf')
    for i in arr5:
        if i>firstlargest:
            secondlargest=firstlargest
            firstlargest=i
        elif i>secondlargest and i!=firstlargest:
            secondlargest=i
    print(firstlargest,secondlargest)
seclarge([10,20,20])

File: day1_js/day1.js
// // create a function to check whether a person is eligilble to vote or not

// function voters(name,age){
//     if(age>=18){
//         return name + " " +"is eligbile for vote"

//     }else{
//         return name + " "+ "not eligible"
//     }

// }
// let voter1=voters("karthilk",20)
// let voter2=voters("kalyan",17)
// console.log(voter1);
// console.log(voter2);


// // write a function that searches for an element in an array and returns the index
// function search(arr,element){
//     for(i=0;i<arr.length;i++){
//         if(arr[i]==element){
            
//             console.log(i);
//         }
        
//     }
//     console.log("-1");
    
// }
// search([1,25, 8,72,100],720)

// // write a function that returns the number of negative numbers in an array
// function num(arr2){
//     count=0
//     for(i=0;i<arr2.length;i++){
//         if(arr2[i]<0){
//             count++
//         }

//     }
//     console.log(count);
    
// }
// num([4,5,6,-1,-5])

// // write a function that returns largest number in an array
// function largest(arr3){
//     large= -Infinity
//     for(i=0;i<arr3.length;i++){
//         if(arr3[i]>large){
//             large=arr3[i]
//         }
//     }
//     console.log(large);
    
// }
// largest([-3,-500])








// let arr=[1,3,5,4,2,7]
// target=8
// for( i=0;i<=arr.length;i++){
//     for (j=0;j<=arr.length;i++){
//         if(arr[i]+arr[j]==target){
//             console.log(arr[i],arr[j]);
            
            
            
//         }
        
//     }
    
// }

// palindrome

// let arr="madam"
// rev=""
// for(i=arr.length-1;i>=0;i--){
//     rev=rev+arr[i]
    
// }
// if(rev==arr){
//     console.log(`given string${arr} is palindrome`);
    
// }else{
//     console.log(`given string ${arr} is not palindrome`);
    
// }


// find given number is palindrome

function ispalindrome(num){
    let originalnum=num
    let rev=0
    while(num!=0){
        rev=rev*10+ num%10
        num=Math.floor(num/10)
    }
    return rev==originalnum
}
console.log(ispalindrome(121));

// find the second largest in an array
function seclarge(arr){
    if(arr.length<2){
        return "array should have 2 elements"
    }
    firstlargest=-Infinity
    secondlargest=-Infinity
    for(let i of arr){
        if(i>firstlargest ){
            secondlargest=firstlargest
            firstlargest=i
            
        }
        else if(i > secondlargest && i!=firstlargest){
            secondlargest=i
        }
    }
    if(secondlargest===-Infinity){
        console.log("no second largest found");
        
    }else{

        console.log(secondlargest);
    }
    

}
seclarge([10,20,20])
